messages/helpers.go

```diff
@@ -141,86 +141,63 @@ func HasUniqueSenders(messages []*proto.Message) bool {
 	return true
 }
 
-// HaveSameProposalHash checks if the messages have the same proposal hash
-func HaveSameProposalHash(messages []*proto.Message) bool {
+// AreValidPCMessages validates PreparedCertificate messages
+func AreValidPCMessages(messages []*proto.Message, height uint64, roundLimit uint64) bool {
 	if len(messages) < 1 {
 		return false
 	}
 
+	round := messages[0].View.Round
+	senderMap := make(map[string]struct{})
```

Comment:
Maybe pre-allocate the worst case scenario (probably a minor thing, but still). WDYT?
```suggestion
	senderMap := make(map[string]struct{}, len(messages))
```